src/frontend/Komponenter/Behandling/VedtakOgBeregning/Opphør/Opphør.tsx

```diff
@@ -0,0 +1,107 @@
+import MånedÅrVelger from '../../../../Felles/Input/MånedÅr/MånedÅrVelger';
+import React, { FormEvent, useState } from 'react';
+import { useBehandling } from '../../../../App/context/BehandlingContext';
+import { Hovedknapp } from 'nav-frontend-knapper';
+import { useApp } from '../../../../App/context/AppContext';
+import { EBehandlingResultat, IOpphørtVedtak, IVedtak } from '../../../../App/typer/vedtak';
+import { Ressurs, RessursStatus } from '../../../../App/typer/ressurs';
+import { useHistory } from 'react-router-dom';
+import { AlertStripeFeil } from 'nav-frontend-alertstriper';
+import styled from 'styled-components';
+import { EnsligTextArea } from '../../../../Felles/Input/TekstInput/EnsligTextArea';
+
+const StyledFormElement = styled.div`
+    margin-bottom: 2rem;
+`;
+
+export const Opphør: React.FC<{ behandlingId: string; lagretVedtak?: IVedtak }> = ({
+    behandlingId,
+    lagretVedtak,
+}) => {
+    const [laster, settLaster] = useState(false);
+    const lagretOpphørtVedtak =
+        lagretVedtak?.resultatType === EBehandlingResultat.OPPHØRT
+            ? (lagretVedtak as IOpphørtVedtak)
+            : undefined;
+    const [opphørtFra, settOpphørtFra] = useState<string>(lagretOpphørtVedtak?.opphørFom || '');
+    const [opphørtBegrunnelse, settOpphørtBegrunnelse] = useState<string>(
+        lagretOpphørtVedtak?.begrunnelse || ''
+    );
+    const [feilmelding, settFeilmelding] = useState<string | undefined>();
+    const { behandlingErRedigerbar, hentBehandling, settAntallIRedigeringsmodus } = useBehandling();
+    const { axiosRequest } = useApp();
+    const history = useHistory();
+
+    const lagreVedtak = (e: FormEvent<HTMLFormElement>) => {
```

Comment:
Litt fundersam på om det er en del kode her til rad 69 som er felles for både opphørt og innvilget?

Reply:
Ja, men forskjellig data-objekt og mye oppdatering av intern state... Med mindre man lager en hook som holder på "samme state" gir det ikke så mye verdi å trekke det ut.